### PR TITLE
brev-cli: init at 0.6.95

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Brev.dev makes it easy to develop on remote machines. 

https://github.com/brevdev/brev-cli

This is the CLI client package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/misc/brev-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "09qhc38ycqs6vdyy1rxaxbihq0q59g2ffkmmlk6ipdlv2b3ywz0f";
+  };
+
+  vendorSha256 = "sha256-5P9oodntXn7RMpjKLoCXlnEZeW4/W0hfYPt7I3hjvGw=";
+
+  CGO_ENABLED = 0;
+  subPackages = [ "." ];
+
+  ldflags = [
+    "-X github.com/brevdev/brev-cli/pkg/cmd/version.Version=${src.rev}"
```
Comment (SuperSandro2000) [inline pkgs/development/misc/brev-cli/default.nix:23]:
```suggestion
    "-s"
    "-w"
    "-X github.com/brevdev/brev-cli/pkg/cmd/version.Version=${src.rev}"
```

File: pkgs/development/misc/brev-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "brev-cli";
+  version = "0.6.95";
+
+  src = fetchFromGitHub {
+    owner = "brevdev";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/development/misc/brev-cli/default.nix:12]:
```suggestion
    repo = "brev-cli";
```

Comment (dit7ya) [reply pkgs/development/misc/brev-cli/default.nix:12]: I am a bit curious about the rationale behind this. Isn't this a pretty common pattern in nixpkgs?

Comment (SuperSandro2000) [reply pkgs/development/misc/brev-cli/default.nix:12]: Yeah, it is pretty common but it is technically not really correct and can cause issues when pname is updated which is arguable only common in pythonPackages.

**Pull request merged**